.github/workflows/coverage.yml

```diff
@@ -1,34 +1,47 @@
 name: Python Coverage
-
-on: [push, pull_request]
-
+on: 
+  push:
+  pull_request:
+  workflow_dispatch:
 jobs:
   build:
-
     runs-on: ubuntu-latest
-
+    timeout-minutes: 3
+    
     steps:
-    - name: Check out code
-      uses: actions/checkout@v2
-
-    - name: Set up Python
-      uses: actions/setup-python@v2
-      with:
-        python-version: '3.x'
-
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements.txt
-
-    - name: Run coverage
-      run: |
-        coverage run -m unittest discover
-        coverage html
-
-    - run: smokeshow upload htmlcov
-      env:
-        SMOKESHOW_GITHUB_STATUS_DESCRIPTION: CLI Coverage {coverage-percentage}
-        SMOKESHOW_GITHUB_COVERAGE_THRESHOLD: 50
-        SMOKESHOW_GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        SMOKESHOW_GITHUB_PR_HEAD_SHA: ${{ github.event.pull_request.head.sha }}
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
```

Comment:
better use @v4

Reply:
Per your request, I updated `actions/checkout@v2` into v4, and also update `actions/setup-python@v2` into v4